onprc_ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest2.java

```diff
@@ -967,6 +968,67 @@ public void testManageCases() throws Exception
         waitForElementToDisappear(Ext4Helper.Locators.window("Manage Cases: " + SUBJECTS[0]));
     }
 
+    @Test
+    public void testFormTriggers()
+    {
+        String aliveAnimal = "99998";
+        String deadAnimal = "99999"; //Mark an animal dead.
+
+        Map<String, Map<String, String>> formsToTest = Map.of(
+                "Birth", Map.of("allowDatesInDistantPast", "true", "allowFutureDates", "true", "allowDeadIds", "true", "allowAnyId", "true"),
+                "Death", Map.of("allowDatesInDistantPast", "true", "allowFutureDates", "true", "allowDeadIds", "true", "allowAnyId", "true"),
+                "Arrival", Map.of("allowDatesInDistantPast", "true", "allowFutureDates", "true", "allowDeadIds", "true", "allowAnyId", "true"),
+                "Departure", Map.of("allowDatesInDistantPast", "true", "allowFutureDates", "true", "allowDeadIds", "true", "allowAnyId", "true"),
+                "Housing Transfers", Map.of("allowDatesInDistantPast", "true", "allowFutureDates", "true", "allowDeadIds", "true", "allowAnyId", "true"));
+
+        goToEHRFolder();
+        clickAndWait(Locator.linkWithText("Enter Data / Task Review"));
+        for (Map.Entry<String, Map<String, String>> form : formsToTest.entrySet())
+        {
+            clickAndWait(Locator.linkWithText(form.getKey()));
+            enableForm();
+            Ext4GridRef grid;
+            if(form.getKey().equals("Housing Transfers"))
+                grid = _helper.getExt4GridForFormSection(form.getKey());
+            else
+                grid = _helper.getExt4GridForFormSection(form.getKey() + "s");
```

Comment:
Like I mentioned above, lets just specify in the initial table what grid to test.

Reply:
Done